lib/routemaster/middleware/siphon.rb

```diff
@@ -0,0 +1,28 @@
+module Routemaster
+  module Middleware
+    class Siphon
+      # Filters out events based on their topic and passes them to a handling class
+      # Usage:
+      #    use Middleware::Siphon, 'some_topic' => SomeTopicSeriesBuilder
+      def initialize(app, processors)
+        @app = app
+        @processors = processors
+      end
+
+      def topics_to_avoid
+        @processors.keys
+      end
+
+      def call(env)
+        time_series, payloads = env.fetch('routemaster.payload', []).partition do |event|
```

Comment:
Maybe use `siphoned` rather than `time_series` as the names as it might not be a time series (I suspect this is left over naming).